pandas/tests/io/test_common.py

```diff
@@ -190,6 +190,25 @@ def test_write_fspath_hdf5(self):
 
         tm.assert_frame_equal(result, expected)
 
+    def test_split_url_extract_uname_pwd(self):
+        """
+        Test extraction of username, pwd from url, if contained.
+        """
+        for url, uname, pwd, nurl in [('https://aaa:bbb@ccc.com:1010/aaa.txt',
```

Comment:
You can take advantage of `pytest` parametrisation decorator.  See <a href="https://docs.pytest.org/en/latest/parametrize.html">here</a> for how to do it.